nautobot/dcim/models/device_components.py

```diff
@@ -519,6 +519,9 @@ def clean(self):
         if not self.mode and self.untagged_vlan is not None:
             raise ValidationError({"untagged_vlan": "Mode must be set when specifying untagged_vlan"})
 
+        if self.tagged_vlans.exists() and self.mode != InterfaceModeChoices.MODE_TAGGED:
+            raise ValidationError({"tagged_vlans": f"Clear tagged_vlans to set mode to {self.mode}"})
```

Comment:
Does this make the `self.tagged_vlans.clear()` in `save()` below now redundant? Should it be removed?